packages/solana/src/client.ts

```diff
@@ -110,13 +111,51 @@ export class Web3Modal extends Web3ModalScaffold {
       },
 
       getApprovedCaipNetworksData: async () =>
-        new Promise(resolve => {
-          const result = {
-            approvedCaipNetworkIds: undefined,
-            supportsAllNetworks: true
-          }
+        new Promise(async resolve => {
+          const walletChoice = localStorage.getItem(SolConstantsUtil.WALLET_ID)
+          if (walletChoice?.includes(ConstantsUtil.WALLET_CONNECT_CONNECTOR_ID)) {
+            const provider = await this.WalletConnectConnector.getProvider()
+            if (!provider) {
+              throw new Error(
+                'networkControllerClient:getApprovedCaipNetworks - connector is undefined'
+              )
+            }
+            const ns = provider?.session?.namespaces
+            const nsChains = ns?.['solana']?.chains
+            const result = {
+              supportsAllNetworks: false,
```

Comment:
Simple if the wallet is not connected with WC; we are setting `supportsAllNetworks` as `true`, and setting it as `true`  if connected with WC no matter what right? Addition to the question in testing, why the Testnet getting not supported? Confused a bit 🙏🏽

Reply:
If connected via WC => depends on the namespaces of the session
If wallet adapter => depends on the wallet's `chains` 
else => there's nothing to compare, we allow all networks (default state on other clients as well)